mvcc/kvstore.go

```diff
@@ -278,9 +278,9 @@ func (s *store) compact(trace *traceutil.Trace, rev int64) (<-chan struct{}, err
 			return
 		}
 		start := time.Now()
-		keep := s.kvindex.Compact(rev)
+		unwanted, unwantedTomb := s.kvindex.Compact2(rev)
```

Comment:
Do we still need `Compact2` when we replace `Compact` here?

Reply:
no need, it's just a mark.

Reply:
I think we might keep two implementations. Because etcd use the old version of compact to compute the hash see function `HashByRev` 
// HashByRev computes the hash of all MVCC revisions up to a given revision.

Reply:
If we change etcd to use the new compact method, shouldn't we use the same compact method to compute HashByRev? HashByRev can be used to validate hash cross different members in the same cluster. I think it needs to reflect how the keys are actually compacted in the store.